### PR TITLE
R refactor

### PR DESCRIPTION
**Pull request by jnolis**
Okay this builds off #293 and rearranges the images so they're all easier to use. I also discovered the CUDA stuff isn't working for torch, so I took that out for now. That means the R images are now:

**Base images**

* `saturnbase-rstudio` - installs R & Rstudio (no JupyterLab)
* `saturnbase-rstudio-gpu-11.2` - installs R & Rstudio (no JupyterLab), built off a rocker/cuda base image
 
**Usable images**

* `saturn-rstudio` - installs a bunch of common R packages
* `saturn-rstudio-lite` - installs nothing. This is mostly for us testing internally, I don't think it needs to be in customer deployments
* `saturn-rstudio-tensorflow` - installs the common R packages plus the ones required to use tensorflow in R

Compared to `release-2021.11.10` this removes `saturn-r` since it's superceded by `saturn-rstudio`

### CODE REVIEW

**Comment by jnolis** [comment]
@forana if you have a chance can you check that `saturnbase-rstudio/Dockerfile` and  `saturnbase-rstudio-gpu-11.1/Dockerfile` look okay? I moved a bunch of things from postScripts to the docker files and I am not positive my `sudo` usage follows best practices

File: saturnbase-rstudio-gpu-11.1/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,167 @@
+FROM rocker/cuda:cuda11.1
+EXPOSE 8888
+
+LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
+      org.opencontainers.image.source="https://github.com/saturncloud/images" \
+      org.opencontainers.image.vendor="Saturn Cloud" \
+      org.opencontainers.image.authors="Jacqueline Nolis" \
+      org.opencontainers.image.description="Additions to rocker/cuda to support running on SaturnCloud"
```
Comment (forana) [inline saturnbase-rstudio-gpu-11.1/Dockerfile:8]:
```suggestion
      org.opencontainers.image.description="Additions to rocker/cuda to support running on Saturn Cloud"
```

File: saturnbase-rstudio-gpu-11.1/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+# ADDITIONAL R SETUP BEYOND BASE IMAGE
+
+RUN sudo apt update && sudo apt-get install -y \
+    devscripts \
+    libxml2 \
+    libxml2-dev \
+    libcurl4-openssl-dev \
+    libssl-dev \
+    libzmq3-dev \
+    qpdf
```
Comment (forana) [inline saturnbase-rstudio-gpu-11.1/Dockerfile:97]: These can get moved up into the main `apt` block

File: saturnbase-rstudio-gpu-11.1/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        libpq5 \
+        libsm6 \
+        libcurl4-openssl-dev \
+        libssl1.1 \
+        openssl \
+        libssl-dev \
+        libuser \
+        libuser1-dev \
+        openssh-client \
+        rrdtool \
```
Comment (forana) [inline saturnbase-rstudio-gpu-11.1/Dockerfile:135]: These can get moved up into the main `apt` block.

File: saturnbase-rstudio-gpu-11.1/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+
+## Install RStudio
+
+ARG RSTUDIO_VERSION=2021.09.0-351
+
+ARG TINI_VERSION=0.18.0
+
+# hadolint ignore=DL3008,DL3009
+RUN sudo apt-get update --fix-missing \
+    && sudo update-locale LANG=en_US.utf8 \
```
Comment (forana) [inline saturnbase-rstudio-gpu-11.1/Dockerfile:121]: Don't need to update again here.

File: saturnbase-rstudio/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+    r-base-dev 
+
+RUN sudo apt-get install -y \
+    devscripts \
+    libxml2 \
+    libxml2-dev \
+    libcurl4-openssl-dev \
+    libssl-dev \
+    libzmq3-dev \
+    qpdf
```
Comment (forana) [inline saturnbase-rstudio/Dockerfile:116]: Can move these up into the first `apt` block.

File: saturnbase-rstudio/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    --no-install-recommends \
+        software-properties-common
+
+RUN sudo su -c \
+    "echo 'deb http://cloud.r-project.org/bin/linux/debian bullseye-cran40/' >>/etc/apt/sources.list"
+
+RUN sudo apt update && sudo apt-get install -y -t bullseye-cran40 \
+    r-base \
+    r-base-core \
+    r-base-dev
```
Comment (forana) [inline saturnbase-rstudio/Dockerfile:107]:
```suggestion
RUN sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-key '95C0FAF38DB3CCAD0C080A7BDC78B2DDEABC47B7' && \
    sudo su -c \
    "echo 'deb http://cloud.r-project.org/bin/linux/debian bullseye-cran40/' >>/etc/apt/sources.list" && \
    sudo apt update && \
    sudo apt-get install -y -t bullseye-cran40 \
        r-base \
        r-base-core \
        r-base-dev 
```

And move `software-properties-common` into the main `apt` block.

File: saturnbase-rstudio/Dockerfile [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+    && sudo mkdir -p /usr/lib/R/site-library \
+    && sudo mkdir -p /usr/lib/R/library \
+    && sudo chown 1000:1000 -R /usr/local/lib/R \
+    && sudo chmod 777 -R /usr/local/lib/R \
+    && sudo chown 1000:1000 -R /usr/lib/R \
+    && sudo chmod 777 -R /usr/lib/R \
+    && sudo su -c "echo 'R_LIBS=/usr/local/lib/R:/usr/local/lib/R/site-library:/usr/lib/R/site-library:/usr/lib/R/library' >> /usr/lib/R/etc/Renviron"
+
+# Setup R reticulate for Python
+RUN sudo su -c "echo 'RETICULATE_PYTHON=/opt/conda/envs/saturn/bin/python' >> /usr/lib/R/etc/Renviron"
```
Comment (forana) [inline saturnbase-rstudio/Dockerfile:130]: Can collapse this `RUN` and the one above it into the `RUN` before them.

File: saturnbase-rstudio/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        libpq5 \
+        libsm6 \
+        libcurl4-openssl-dev \
+        libssl1.1 \
+        openssl \
+        libssl-dev \
+        libuser \
+        libuser1-dev \
+        openssh-client \
+        rrdtool \
```
Comment (forana) [inline saturnbase-rstudio/Dockerfile:154]: Can move these into the main `apt` block.

**Comment by forana** [comment]
Code looks good - setting up a test build of `saturn-rstudio`.

**Comment by jnolis** [comment]
> Code looks good - setting up a test build of `saturn-rstudio`.

Should we test `saturn-rstudio-tensorflow` too, since they have different base images?

**Comment by forana** [comment]
> Should we test saturn-rstudio-tensorflow too, since they have different base images?

Yes, but I'm most worried about rstudio, since this build will replace the one we've released and tested once we merge and do a proper build.

**Review by forana — APPROVED** [review]
LGTM

**Pull request merged**